test.py

```
@@ -517,53 +517,78 @@ def __init__(self, test_no, shortname, suite):
-            shutil.copyfile(self.tmpfile, self.reject)
-        elif os.path.exists(self.tmpfile):
-            pathlib.Path(self.tmpfile).unlink()
+        async with self.suite.clusters.instance() as cluster:
+            self.args.insert(1, "--host={}".format(cluster[0].host))
+            # If pre-check fails, e.g. because Scylla failed to start
+            # or crashed between two tests, fail entire test.py
+            try:
+                cluster.pre_check()
+                self.is_pre_check_ok = True
+                cluster[0].take_log_savepoint()
+                self.is_executed_ok = await run_test(self, options, env=self.env)
+                cluster.post_check()
+                self.is_post_check_ok = True
+
+                if self.is_executed_ok is False:
+                    set_summary("""returned non-zero return status.\n
+Check test log at {}.""".format(self.log_filename))
+                elif not os.path.isfile(self.tmpfile):
+                    set_summary("failed: no output file")
+                elif not os.path.isfile(self.result):
+                    set_summary("failed: no result file")
+                    self.is_new = True
+                else:
+                    self.is_equal_result = filecmp.cmp(self.result, self.tmpfile)
+                    if self.is_equal_result is False:
+                        self.unidiff = format_unidiff(self.result, self.tmpfile)
+                        set_summary("failed: test output does not match expected result")
+                        logging.info("\n{}".format(palette.nocolor(self.unidiff)))
+                    else:
+                        self.success = True
+                        set_summary("succeeded")
+            except Exception as e:
+                # Server log bloats the output if we produce it in all
+                # cases. So only grab it when it's relevant:
+                # 1) failed pre-check, e.g. start failure
+                # 2) failed test execution.
```

I think you should check **post**-check, not **pre**-check.
In other words, if Scylla crashed during a specific test - even if the test itself passed miraculously (?) - we should report this failure for **that** test. I don't want to see a failure report for the **next** test just because when it started, its pre-check failed. That is very confusing - you see a failure in test "xyz" but after a lot of thinking you realize there is nothing wrong with test xyz, it was actually the test abc that ran just before it crashed Scylla, and xyz couldn't even start.

---

If scylla crashed during the test, post_check() will detect it because it runs a query against scylla - getting the keyspace count. This will be correctly attributed to the test which caused the crash.